lib/gst/player/gstplayer.c

```diff
@@ -3179,6 +3180,84 @@ gst_player_set_visualization_enabled (GstPlayer * self, gboolean enabled)
       enabled ? "Enabled" : "Disabled");
 }
 
+/**
+ * gst_player_get_color_balance_list_channels:
+ * @player: #GstPlayer instance
+ *
+ * Retrieve a list of the available channels.
+ *
+ * Returns: (element-type Gst.ColorBalanceChannel) (transfer none): A
+ *          GList containing pointers to #GstColorBalanceChannel
+ *          objects. The list is owned by the #GstColorBalance
+ *          instance and must not be freed.
+ */
+const GList *
+gst_player_get_color_balance_list_channels (GstPlayer * self)
+{
+  g_return_val_if_fail (GST_IS_PLAYER (self), NULL);
+
+  if (!GST_IS_COLOR_BALANCE (self->playbin))
```

Comment:
Good point. Yet another reason to hide all this GstColorBalance stuff more from the application and do all the difficult things internally :)